lib/cli/src/Cardano/CLI.hs

```diff
@@ -630,38 +638,92 @@ cmdWalletUpdateName mkClient =
             (ApiT wId)
             (WalletPutData $ Just (ApiT wName))
 
+data UpdatePassphraseCredential = MnemonicCredentials | OldPasswordCredentials
+    deriving Eq
+
+-- | Which json schema to use for output
+useMnemonicOption :: Parser UpdatePassphraseCredential
+useMnemonicOption = flag OldPasswordCredentials MnemonicCredentials
+    $ long "mnemonic"
+        <> help "use mnemonic to authorize the passphrase change"
+
 -- | Arguments for 'wallet update passphrase' command
 data WalletUpdatePassphraseArgs = WalletUpdatePassphraseArgs
     { _port :: Port "Wallet"
     , _id :: WalletId
+    , _mnemonic :: UpdatePassphraseCredential
     }
 
-cmdWalletUpdatePassphrase
-    :: ToJSON wallet
-    => WalletClient wallet
-    -> Mod CommandFields (IO ())
-cmdWalletUpdatePassphrase mkClient =
-    command "passphrase" $ info (helper <*> cmd) $ mempty
-        <> progDesc "Update a wallet's passphrase."
-  where
-    cmd = fmap exec $ WalletUpdatePassphraseArgs
-        <$> portOption
-        <*> walletIdArgument
-    exec (WalletUpdatePassphraseArgs wPort wId) = do
-        res <- sendRequest wPort $ getWallet mkClient $ ApiT wId
-        case res of
-            Right _ -> do
-                wPassphraseOld <- getPassphrase
-                    "Please enter your current passphrase: "
-                wPassphraseNew <- getPassphraseWithConfirm
-                    "Please enter a new passphrase: "
-                runClient wPort (const mempty) $
-                    putWalletPassphrase mkClient (ApiT wId) $
-                        WalletPutPassphraseData
-                            (ApiT wPassphraseOld)
-                            (ApiT wPassphraseNew)
-            Left _ ->
-                handleResponse Aeson.encodePretty res
+class CmdWalletUpdatePassphrase wallet where 
+    cmdWalletUpdatePassphrase
+        :: ToJSON wallet
+        => WalletClient wallet
+        -> Mod CommandFields (IO ())
+instance CmdWalletUpdatePassphrase ApiWallet where
+    cmdWalletUpdatePassphrase mkClient =
+        command "passphrase" $ info (helper <*> cmd) $ mempty
+            <> progDesc "Update a wallet's passphrase."
+        where
+            cmd = fmap exec $ WalletUpdatePassphraseArgs
+                <$> portOption
+                <*> walletIdArgument
+                <*> useMnemonicOption
```

Comment:
Alternatively:
```suggestion
            cmd = exec <$>
                liftA2 ByronWalletUpdatePassphraseArgs
                    portOption walletIdArgument
```

Reply:
I tend to use the long boring <$> .. <*> .. as universal. Is this acceptable?

Reply:
> I tend to use the long boring <$> .. <*> .. as universal. Is this acceptable?

Yes! I prefer it to `liftA2`, actually, because it is more regular.